onnx/defs/schema.cc

```diff
@@ -499,13 +499,19 @@ void OpSchema::Finalize() {
   for (const auto& it : outputs_) {
     ENFORCE(!(it.GetName().empty()));
   }
+
+  // Only the last input could be variadic.
+  for (int i = 0; i < (int)(inputs_.size()) - 1; ++i) {
+    ENFORCE(Variadic != inputs_[i].GetOption());        
+  }
+
   // TODO: also cover checks for arbitrary number of inputs
   // allow extra tailing inputs not be present if all inputs at the end are
   // marked as optional
   if (max_input_ < std::numeric_limits<int>::max()) {
     int ind = max_input_;
     for (auto& input : inputs_) {
```

Comment:
this code should probably also look at indices from the end only. Original (my) implementation was not very correct

Reply:
will revisit this logic in separate PR together with generating min/max input/output without NumInputs, NumOutputs specification.
Basically, min is the number of "Single"s. max is sum of number of singles and optionals. if there's variadic, max is INT_MAX.
Make sense?